src/Compilers/Core/Desktop/AnalyzerFileReference.InMemoryAssemblyLoader.cs

```diff
@@ -202,6 +202,12 @@ private static Assembly ResolveForUnknownRequestor(string requestedAssemblyName)
                     foreach (string loadedAssemblyFullPath in s_assembliesFromFiles.Keys)
                     {
                         string directoryPath = Path.GetDirectoryName(loadedAssemblyFullPath);
+
+                        if (!string.IsNullOrEmpty(requestedAssemblyIdentity.CultureName))
+                        {
+                            directoryPath = Path.Combine(directoryPath, requestedAssemblyIdentity.CultureName);
```

Comment:
On the LCID thing - will the CLR assembly resolution for normal contexts pick up a Japanese resource if it was in the 1041 directory for example? If so, yes we should look for the LCID directory - otherwise I don't think there'll be a scenario where the user will expect it.